syntax = "proto3";

package hotbox;

message ParserConfig {
  oneof config {
    LibSVMParserConfig libsvm_config = 2;
    CSVParserConfig csv_config = 1;
  }
}

message LibSVMParserConfig {
  // True if feature indices starts from 1.
  bool feature_one_based = 1;

  // True if label indices starts from 1.
  bool label_one_based = 2;
}

message CSVParserConfig {
  // the positon of the column name, 0 means no column name.
  int32 header_position = 1;

  // True if label indices starts from 0.
  bool first_column_label = 2;
 // bool label_end = false;
}
